spec/spec.md

```diff
@@ -18,20 +18,12 @@ integration and serving the needs of product teams with less knowledge, requirem
 
 ### Definitions
 
-- base asset: The asset that the user needs to know the value or price for (e.g: USDC as in "I need to know the value of
-  1e6 USDC in ETH terms").
-- quote asset: The asset in which the user needs to value or price the `base` (e.g: ETH as in "I need to know the value
-  of 1e6 USDC in ETH terms").
-- unit: Minimum representable amount of an asset on-chain.
-- decimals: The number of positions to move the comma to the left to make a `whole unit` out of a unit (e.g. USDC has 6
-  `decimals`).
-- scalar: 10 to the power of `decimals` (e.g. USDC's `scalar` is 1e6).
-- whole unit: A `scalar` amount of `unit` (e.g. A `whole unit` of USDC is 1e6 `units`).
+- base asset: The asset that the user needs to know the value for (e.g: USDC as in "I need to know the value of 1e6 USDC
+  in ETH terms").
+- quote asset: The asset in which the user needs to value the `base` (e.g: ETH as in "I need to know the value of 1e6
+  USDC in ETH terms").
 - value: An amount of `base` in `quote` terms (e.g. The `value` of 1000e6 USDC in ETH terms is 283,969,794,427,307,000
   ETH, and the `value` of 1000e18 ETH in USDC terms is 3,521,501,299,000 USDC).
```

Comment:
```suggestion
- base asset: The asset that the user needs to know the value for (e.g: USDC as in "I need to know the value of 1e6 USDC
  in ETH terms").
- quote asset: The asset in which the user needs to value the `base` (e.g: ETH as in "I need to know the value of 1e6
  USDC in ETH terms").
- value: An amount of `base` in `quote` terms (e.g. The `value` of 1000e6 USDC in ETH terms is 283,969,794,427,307,000
  ETH, and the `value` of 1000e18 ETH in USDC terms is 3,521,501,299,000 USDC).
- decimals: The decimals for a given asset as defined in ERC20. Throughout the spec, `base` and `quote` asset amounts are denoted using their respective decimals.
```

while it may seem obvious, i see value in clarifying early on how the oracles process decimals. in particular, explicitly pointing out that `base` and `quote` asset amounts are always denoted using their respective decimals.

Reply:
I'm a bit confused, decimals are not used in the spec, only absolute amounts such as 1e6 USDC and 1000e18 ETH.

Reply:
agreed, the spec does not use decimals so this doesn't belong in the definitions. let me try to reframe my concern

---

consider two cases:
1. an oracle that complies with `IOracle` but always returns 18 decimal return values
2. an oracle that complies with `IOracle` but defines a custom precision for return values

imo we should explicitly discourage such implementations and define it such that they are non-compliant. to do so we must specify that `baseAmount` and `quoteAmount` are denoted in the same decimals as `base` and `quote` respectively. this would remove any ambiguity.

i think adding a statement to the tune of "throughout the spec `baseAmount` and `quoteAmount` are assumed to have the same precision as `base` and `quote` respectively" achieves this.

Reply:
To comply with `IOracle` any oracle needs to return integer token amounts, anything else is not compliant. Added an extra line to be crystal clear.